Frontend/src/pages/SignUpPage.jsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import { useState } from 'react';
+import { Bounce, ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import CssBaseline from '@mui/material/CssBaseline';
+import TextField from '@mui/material/TextField';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Link from '@mui/material/Link';
+import Paper from '@mui/material/Paper';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import { useNavigate } from 'react-router-dom';
+import axios from 'axios';
+
+
+function SignUpPage() {
+
+
+
+  const navigate = useNavigate();
+
+  // States for form fields and validation errors
+  const [formFields, setFormFields] = useState({ username: '', password: '' });
+  const [formErrors, setFormErrors] = useState({ username: '', password: '' });
+
+  // Handle real-time field value changes and validate
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormFields({ ...formFields, [name]: value });
+
+    // Validate fields
+    if (name === 'username') {
+      setFormErrors({
+        ...formErrors,
+        username: value.length >= 3 && value.length <= 50 ? '' : 'Username must be between 3 and 50 characters',
+      });
+    } else if (name === 'password') {
+      setFormErrors({
+        ...formErrors,
+        password: value.length >= 6 && value.length <= 50 ? '' : 'Password must be between 6 and 50 characters',
+      });
+    }
```

Comment:
What? why comparing the name to username? 
If the logic is right, extract to a function that her name is what you're trying to do 
validateUsernameAndPasswordAreDefault or something like this to reflect this